407 enhancement deprecation warnings for pastasread methods

OnnoEbbens opened the pull request:
# Short Description
Add a deprecation warning for all functions in pastas.read. Modify the notebook of Dutch datasets to use Hydropandas.

# Checklist before PR can be merged:
- [x] closes issue #407
- [x] PEP8 compliant code
- [x] Example Notebook (for new features)

OnnoEbbens commented:
I have added an example of how you can use Hydropandas to create a Pastas model using Dutch datasets. I think pastas tests all notebooks so maybe this means we have to install hydropandas for the tests. If that is something we don't want I can move the notebook to hydropandas.

raoulcollenteur commented:
I think we can remove the read_dutch_data notebook, and as we don't want to add a dependency to HydroPandas, so all the example data needs to be replaced or read using pandas.read_csv. Quite a bit of work actually...

OnnoEbbens commented:
You are right. I did the following:
- remove the Dutch dataset notebook. I added all the information in the notebook to a hydropandas notebook -> https://github.com/ArtesiaWater/hydropandas/blob/724efed5aa6fbd0780788611a0b35ea86f3fc4ee/examples/03_hydropandas_and_pastas.ipynb
- I removed all Dino, KNMI, Waterbase and Menyanthes files and replaced them with csv files. This also involved some small changes to the notebooks. I visually checked if the results remained the same.
- remove the tests specifically testing the old read functions. I think it is not necessary to replace these tests with .csv files because the Pastas functions used in these tests are also tested elsewhere.

I still want to add the example of reading a Menyanthes file to hydropandas. This is not related to this pull request

OnnoEbbens commented:
I have no idea what is happening to the tests. It seems like this pull_requests is not tested again when I commit to it but it does show the failure of the last test.

raoulcollenteur commented on requirements.ci.txt:
```diff
@@ -10,4 +10,6 @@ jupyter>=1.0.0
\ No newline at end of file
+lmfit>=1.0.0
+
+hydropandas>=0.6.0
```
This is not necessary anymore right now I think?

raoulcollenteur reviewed (changes requested):
Can you remove hydropandas from requirements.ci.txt ? Then it should all be good.

raoulcollenteur commented:
That looks like it was quite some work, but great PR! Just one minor comment before "squash and merge"!

OnnoEbbens commented:
Thanks! I removed the hydropandas requirement and yes the PR was more work than expected. Also it is remarkable how often we use the same time series in the example notebooks. Maybe we could try to use more different datasets in the notebooks so we automatically test more edge cases. However that's kind of a boring job to do so only for another lockdown I would say :)
